FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,39 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+
+			switch result {
+			case .success(let (data, httpResponse)):
+				if httpResponse.statusCode == 200, let root = try? JSONDecoder().decode(Root.self, from: data) {
+					completion(.success(root.items.map { $0.item }))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
Consider moving the mapping logic to a helper function to improve readability in the `load` function:
```suggestion
				completion(RemoteFeedLoader.map(data, from: httpResponse))
```